<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ListTaskWindow</class>
 <widget class="QDialog" name="ListTaskWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="mainTabWidget">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="firstTaskTab">
      <attribute name="title">
       <string>Задание 1</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Вокруг считающего стоит N человек, из которых выделен первый, а остальные занумерованы  по часовой стрелке числами от 2 до N. Считающий, начиная с кого-то, ведет счет до M. Человек, на котором остановился счет, выходит из круга. Счет продолжается со следующего человека и так до тех пор, пока не останется один человек. Определить номер оставшегося человека, если известно M и то, что счет начинался с первого человека</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTabWidget" name="InputFirstTaskTabWidget">
         <property name="currentIndex">
          <number>1</number>
         </property>
         <widget class="QWidget" name="josephusManualTab">
          <attribute name="title">
           <string>Вручную</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,0">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <item>
                <widget class="QLineEdit" name="inputManualJosephusLineEdit">
                 <property name="placeholderText">
                  <string>Введите фамилии людей через пробел</string>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout">
                 <item>
                  <widget class="QLabel" name="josephusManualLabel">
                   <property name="text">
                    <string>Шаг считающего</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="josephusManualSpinBox"/>
                 </item>
                </layout>
               </item>
               <item>
                <spacer name="verticalSpacer">
                 <property name="orientation">
                  <enum>Qt::Orientation::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>40</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QPushButton" name="inputJosephusManualPushButton">
                 <property name="text">
                  <string>Ввести</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QPlainTextEdit" name="outputJosephusManualPlainTextEdit">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="clearJosephusManualPushButton">
               <property name="text">
                <string>Очистить вывод</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="josephusFileTab">
          <attribute name="title">
           <string>Из файла</string>
          </attribute>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_6">
               <item>
                <widget class="QLineEdit" name="inputFileJosephusLineEdit">
                 <property name="placeholderText">
                  <string>Введите название файла</string>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_5">
                 <item>
                  <widget class="QLabel" name="josephusFileLabel">
                   <property name="text">
                    <string>Шаг считающего</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="josephusFileSpinBox"/>
                 </item>
                </layout>
               </item>
               <item>
                <spacer name="verticalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Orientation::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>40</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QPushButton" name="inputJosephusFilePushButton">
                 <property name="text">
                  <string>Ввести</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QPlainTextEdit" name="outputJosephusFilePlainTextEdit">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="clearJosephusFilePushButton">
               <property name="text">
                <string>Очистить вывод</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="SecondTaskTab">
      <attribute name="title">
       <string>Задание 2</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Pадан набор целых чисел. Записать их в память используя свой список. Просматривая элементы списка по тройкам, поменять крайние, результат записать в новый список</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTabWidget" name="inputSecondTaskTabWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="swapManualTab">
          <attribute name="title">
           <string>Вручную</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLineEdit" name="inputSwapManualLineEdit">
               <property name="placeholderText">
                <string>Введите числа через пробел</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="inputSwapManualPushButton">
               <property name="text">
                <string>Ввести</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="outputSwapManualPlainTextEdit"/>
           </item>
           <item>
            <widget class="QPushButton" name="cleanSwapManualPushButton">
             <property name="text">
              <string>Очистить вывод</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="swapFileTab">
          <attribute name="title">
           <string>Из файла</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QLineEdit" name="inputSwapFileLineEdit">
               <property name="placeholderText">
                <string>Введите название файла</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="inputSwapFilePushButton">
               <property name="text">
                <string>Ввести</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_9">
             <item>
              <widget class="QPlainTextEdit" name="inputSwapFilePlainTextEdit">
               <property name="placeholderText">
                <string>Тут появится информация из входного файла</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_9">
               <item>
                <widget class="QPlainTextEdit" name="outputSwapFilePlainTextEdit"/>
               </item>
               <item>
                <widget class="QPushButton" name="clearSwapFilePushButton">
                 <property name="text">
                  <string>Очистить вывод</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
